Quantum-Control-Applications/Optically addressable spin qubits/NV center in a confocal setup/09b_xy8_tau.py

```diff
@@ -0,0 +1,217 @@
+
+    play("x180", "NV")  # 8 X
+
+
+###################
+# The QUA program #
+###################
+with program() as xy8_tau:
+    tau = declare(int)  # interpulse spacing
+    tau_half = declare(int)  # half interpulse spacing
+    n = declare(int)  # for averages
+    i = declare(int)  # for xy8 order
+
+    counts1 = declare(int)  # saves number of photon counts
+    counts2 = declare(int)  # saves number of photon counts
+    times1 = declare(int, size=100)  # QUA vector for storing the time-tags
+    times2 = declare(int, size=100)  # QUA vector for storing the time-tags
+    counts_1_st = declare_stream()  # stream for counts
+    counts_2_st = declare_stream()  # stream for counts
+    n_st = declare_stream()  # stream to save iterations
+
+    # Spin initialization
+    play("laser_ON", "AOM1")
+    wait(wait_for_initialization * u.ns, "AOM1")
+
+    # XY8-n sequence
+    with for_(n, 0, n < n_avg, n + 1):
+        with for_(*from_array(tau, tau_vec)):
+            assign(tau_half, tau / 2)
+            with strict_timing_():
+                # First XY8 sequence with x90 - XY8 block - x90
+                play("x90", "NV")
+                xy8_n(xy8_order)
+                play("x90", "NV")
+            align()  # Play the laser pulse after the XY8 sequence
+            # Measure and detect the photons on SPCM1
+            play("laser_ON", "AOM1")
+            measure("readout", "SPCM1", None, time_tagging.analog(times1, meas_len_1, counts1))
+            save(counts1, counts_1_st)  # save counts
+            wait(wait_between_runs * u.ns, "AOM1")
+
+            align()
+            # Second XY8 sequence with x90 - XY8 block - -x90
+            with strict_timing_():
+                # First XY8 sequence with x90 - XY8 block - x90
+                play("x90", "NV")
+                xy8_n(xy8_order)
+                play("-x90", "NV")
+            align()  # Play the laser pulse after the Echo sequence
+            # Measure and detect the photons on SPCM1
+            play("laser_ON", "AOM1")
+            measure("readout", "SPCM1", None, time_tagging.analog(times2, meas_len_1, counts2))
+            save(counts2, counts_2_st)  # save counts
+            wait(wait_between_runs * u.ns, "AOM1")
+
+            wait(wait_between_runs * u.ns, "AOM1")
+
+        save(n, n_st)  # save number of iterations inside for_loop
+
+    with stream_processing():
+        # Cast the data into a 1D vector, average the 1D vectors together and store the results on the OPX processor
+        counts_1_st.buffer(len(tau_vec)).average().save("counts1")
+        counts_2_st.buffer(len(tau_vec)).average().save("counts2")
+        n_st.save("iteration")
+
+#####################################
+#  Open Communication with the QOP  #
+#####################################
+qmm = QuantumMachinesManager(host=qop_ip, cluster_name=cluster_name, octave=octave_config)
+
+#######################
+# Simulate or execute #
+#######################
+simulate = True
```

Comment:
I would set the default to False so that you don't have to always toggle it when running the script during an installation